packages/itwin/measure-tools/src/api/Measurement.ts

```diff
@@ -257,6 +297,10 @@ export abstract class Measurement {
   private _transientId?: Id64String; // Not serialized
   private _isVisible: boolean; // Not serialized
 
+  // Used for sheet measurements
+  private _drawingMetaData?: DrawingMetaData;
+  private _sheetViewId?: string;
```

Comment:
Yes, the information is in drawing coordinates because it's to help locate the drawing in the sheet: 
The origin of the drawing in the sheet and the extent of the drawing in the sheet.
To me, it would not make sense to put it in the drawingMetadata since there can be multiple drawings in the sheet and we often compare multiple drawings that are in the same sheet.
Also, they serve a different purpose and are in different places in the code. The sheetId is mostly used to only display the measurement in the sheet it was created for and the DrawingMetadata is used for the drawing creation and transformation.

Reply:
> To me, it would not make sense to put it in the drawingMetadata since there can be multiple drawings in the sheet and we often compare multiple drawings that are in the same sheet.
The measurement will always be contained within a single drawing on the sheet. It makes sense to keep the sheetId alongside this data. If the same drawing were to be in a different sheet, the placement wouldn't be the same.

> So a `drawingId` is basically a child of a `sheetViewId`? Maybe then the metadata could have a `parentSheetViewId`?
Not a child, it's a reference. This is similar to the concept of geometry parts. A drawing is a container for 2d graphics, and you can reference the drawing as a whole inside a sheet by setting up proper placement (origin, scale). Thus, multiple sheets could reference the same drawing with different combinations of (origin, scale).
I haven't seen a case where a sheet references the SAME drawing more than once, but I guess that would be technically possible too ??

Reply:
> The measurement will always be contained within a single drawing on the sheet. It makes sense to keep the sheetId alongside this data. If the same drawing were to be in a different sheet, the placement wouldn't be the same.

But there could be cases where we have drawings in sheets but not within any drawing boundaries and we still want to hide them when inside the wrong sheet.

Reply:
Meaning we still need to have the sheetId even though the drawingMetadata has not been set

Reply:
That's a good point...
Back to the drawing board -- I wonder if we could augment the MeasurementViewTarget class to support targeting a specific model.
Up until now we only cared about the 'type' of view the measurement lies in. That makes sense for 3d views since they display [0..n] models, but 2d views only display a single model. That could even open the floor to measurements that are bound to particular model(s) in 3d where if you turn one of them off the measurement is hidden.
@bsy-nicholasw

Reply:
You should be able to create a custom viewtype classifier that identifies based on the model with the current design. The typename may be a little flakey though, as you'd probably need to encode it like: `sheetdrawing::<drawing_model_id>`